authors/apps/articles/views.py

```diff
@@ -0,0 +1,105 @@
+from django.shortcuts import render
+
+from rest_framework.generics import (
+    ListCreateAPIView, RetrieveUpdateDestroyAPIView)
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+
+
+from authors.apps.core.permissions import IsOwnerOrReadOnly
+
+from .models import Article
+from .serializers import ArticleSerializer
+from .renderers import ArticleJsonRenderer
+
+
+def get_article(slug):
+
+    try:
+        article = Article.objects.get(slug=slug)
+
+    except Exception:
+        return Response(
+            {"error": "Article not found"},
+            status=status.HTTP_404_NOT_FOUND
+        )
+    return article
+
+
+class ArticleList(ListCreateAPIView):
+    """Allows user to post and view a list of articles"""
+
+    serializer_class = ArticleSerializer
+    renderer_classes = (ArticleJsonRenderer,)
+    queryset = Article.objects.all()
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+
+    def post(self, request):
```

Comment:
This should be in `ArticleDetail`, the view should be for getting a list of articles. You can use regular `APIView` in `ArticleDetail` to have all the functionality.

Reply:
can you please add more color on this one.

Reply:
The `ArticleList` class should be doing only listing(returning many articles). Article creation is an operation to do with the details of the article and thus should be in the `ArticleDetail` view. You can inherit from `APIView` instead of `RetrieveUpdateDestroyAPIView`.

Reply:
Hi, because in the `ArticleDetail` class I am just getting, updating and/or deleting the article using its unique slug, I therefore, cannot add a post method. alternatively, i have created new class for posting articles `ArticleCreate`. Just need clarification if this is a rational enough implementation of the CRUD